src/main/java/org/jmxtrans/embedded/EmbeddedJmxTrans.java

```diff
@@ -248,18 +244,8 @@ public synchronized void stop() throws Exception {
             return;
         }
         try {
-            collectScheduledExecutor.shutdown();
-            try {
-                collectScheduledExecutor.awaitTermination(getQueryIntervalInSeconds(), TimeUnit.SECONDS);
```

Comment:
Must say I prefer the previous implementation here, better to give running tasks a chance to finish.